src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs

```diff
@@ -220,22 +233,21 @@ internal void SetAttributes(string path, FileAttributes attributes, bool asDirec
                 throw new ArgumentException(SR.Arg_InvalidFileAttrs, "Attributes");
             }
 
-            EnsureCachesInitialized(path);
+            EnsureCachesInitialized(handle, path);
 
             if (!EntryExists)
-                FileSystemInfo.ThrowNotFound(path);
+                FileSystemInfo.ThrowNotFound(path ?? "<<handle>>");
```

Comment:
We don't want to throw exceptions with messages where the path is replaced with some magic string. It looks ugly, like `Could not find file '<handle>'.` 

Instead you should pass null, and the callee should gracefully handle this and use an appropriate message.

If yo'ure passing a `ReadOnlySpan<char>` you can't pass null but you can pass `""`. For example when you call `ThrowOnCacheInitializationError`. It looks like that will end up in GetExceptionForIoErrno which has different messages when it does have a path and when it doesn't -- it checks `string.IsNullOrEmpty(..)` so your empty string would lead to a message without a path. See code here:

eg here https://github.com/danmoseley/runtime/blob/78bf14a4111a0b6760a2321b6f8759c83674b216/src/libraries/Common/src/Interop/Unix/Interop.IOErrors.cs#L120

Please ensure you have a test for each of these 3 places where currently you pass `<<handle>>`.